### PR TITLE
SBERDOMA-421 Implement building type validation

### PR DESCRIPTION
**Pull request by MrFoxPro**

### CODE REVIEW

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 1]
```diff
@@ -31,11 +33,15 @@ const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
...
     const PromptTitle = intl.formatMessage({ id: 'pages.condo.property.warning.modal.Title' })
     const PromptHelpMessage = intl.formatMessage({ id: 'pages.condo.property.warning.modal.HelpMessage' })
+    const AddressValidationErrorMsg = intl.formatMessage({ id: 'pages.condo.property.warning.modal.AddressValidationErrorMsg' })
+
 
     const { addressApi } = useAddressApi()
 
     const { action, initialValues } = props
 
+    const restrictedPropertyTypes: AddressMeta['data']['house_type'][] = ['д', 'к']
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:43]: It's better to extract a MAGIC_CONST

File: apps/condo/domains/common/utils/addressApi/AddressMeta.ts [hunk 2]
```diff
@@ -125,7 +50,7 @@ type AddressMeta = {
         street: string;
         house_fias_id: string;
         house_kladr_id: string;
-        house_type: string;
+        house_type: 'д' | 'влд' | 'к';
```
Comment (pahaz) [inline apps/condo/domains/common/utils/addressApi/AddressMeta.ts:53]: it's better to extract MAGIC_CONST

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-import { Col, Form, Input, notification, Row, Typography } from 'antd'
+import { Col, Form, Input, notification, Row, Typography, FormItemProps } from 'antd'
 import React, { useCallback } from 'react'
 import { IPropertyFormState } from '@condo/domains/property/utils/clientSchema/Property'
 import { FormWithAction } from '@condo/domains/common/components/containers/FormList'
 import { AddressSuggestionsSearchInput } from '@condo/domains/property/components/AddressSuggestionsSearchInput'
 import { useAddressApi } from '@condo/domains/common/components/AddressApi'
 import { PropertyPanels } from '../panels'
 import Prompt from '@condo/domains/common/components/Prompt'
+import { AddressMeta } from '../../../common/utils/addressApi/AddressMeta'
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:12]: @condo/common

File: apps/condo/domains/common/utils/addressApi/index.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import getConfig from 'next/config'
```
Comment (pahaz) [inline apps/condo/domains/common/utils/addressApi/index.ts:1]: why you want to make a folder?

File: apps/condo/domains/property/constants/property.ts [hunk 5]
```diff
@@ -1,3 +1,8 @@
+import { AddressMeta } from '@condo/domains/common/utils/addressApi/AddressMeta'
+
+// "д" - "дом" or "к" - "корпус"
+const validHouseTypes: AddressMeta['data']['house_type'][] = ['д', 'к', 'стр']
```
Comment (MrFoxPro) [inline apps/condo/domains/property/constants/property.ts:4]: Should I rename all constants in UPPER_CASE style?

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/property/constants/property.ts (hunk 5 above)
Comment (pahaz) [reply apps/condo/domains/property/constants/property.ts:4]: yes

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 6]
```diff
@@ -9,7 +9,9 @@ import { AddressSuggestionsSearchInput } from '@condo/domains/property/component
 import { useAddressApi } from '@condo/domains/common/components/AddressApi'
 import { PropertyPanels } from '../panels'
 import Prompt from '@condo/domains/common/components/Prompt'
-
+import { AddressMeta } from '@condo/domains/common/utils/addressApi/AddressMeta'
+import { useState } from 'react'
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:13]: it's better to reorder and regroup imports

File: apps/condo/domains/property/components/BasePropertyForm/index.tsx [hunk 7]
```diff
@@ -31,11 +33,15 @@ const BasePropertyForm: React.FC<IPropertyFormProps> = (props) => {
     const AddressMetaError = intl.formatMessage({ id: 'errors.AddressMetaParse' })
     const PromptTitle = intl.formatMessage({ id: 'pages.condo.property.warning.modal.Title' })
     const PromptHelpMessage = intl.formatMessage({ id: 'pages.condo.property.warning.modal.HelpMessage' })
+    const AddressValidationErrorMsg = intl.formatMessage({ id: 'pages.condo.property.warning.modal.AddressValidationErrorMsg' })
+    const UnsupportedPropertyErrorMsg = intl.formatMessage({ id: 'pages.condo.property.warning.modal.UnsupportedPropertyErrorMsg' })
+
```
Comment (pahaz) [inline apps/condo/domains/property/components/BasePropertyForm/index.tsx:38]: redundant space

**Pull request merged**